channels_redis/core.py

```diff
@@ -169,22 +170,14 @@ def _setup_encryption(self, symmetric_encryption_keys):
                         # Raise our own exception if that failed
                         raise RuntimeError("Redis receive loop exited early")
                 # Wait for our message to appear
-                # TODO: Queues, not polling
                 while True:
-                    if self.receive_buffer.get(channel, None):
-                        # There's a message, pop it off and shorten the buffer
-                        message = self.receive_buffer[channel][0]
-                        if len(self.receive_buffer[channel]) == 1:
-                            del self.receive_buffer[channel]
-                        else:
-                            self.receive_buffer[channel] = self.receive_buffer[channel][1:]
-                        return message
-                    else:
+                    try:
+                        return await asyncio.wait_for(self.receive_buffer[channel].get(), self.local_poll_interval)
```

Comment:
You need to clean up the entries in the dict if the queue is empty, or this will memory leak over time.